src/main/java/org/broadinstitute/dsm/jobs/UPSTrackingJob.java

```diff
@@ -21,60 +22,230 @@
+            }
+            catch (Exception e) {
+                dbVals.resultException = e;
+            }
+            return dbVals;
+        });
+        if (result.resultException != null) {
+            throw new RuntimeException(result.resultException);
+        }
+        else {
+            if (!isReturn) {
+                //if delivered notif pepper
+                if (DELIVERY.equals(statusType) && !(DELIVERY.equals(oldType))) {
+                    KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER + SELECT_BY_TRACKING_NUMBER, new String[] { DELIVERED, trackingId }, 2);//todo change this to the number of subkits but for now 2 for test boston works
+                    if (kitDDPNotification != null) {
+                        logger.info("Triggering DDP for delivered kit with external order number: " + kit.getExternalOrderNumber());
+                        EventUtil.triggerDDP(kitDDPNotification);
+
+                    }
+                    else {
+                        logger.error("delivered kitDDPNotification was null for " + kit.getExternalOrderNumber());
+                    }
+                }
 
             }
-            catch (IOException e) {
-                throw new RuntimeException("couldn't get response from ups tracking ", e);
+            else {
+                //if picked up place order
+                if (!(PICKUP.equals(oldType)) && !(IN_TRANSIT.equals(oldType)) && (PICKUP.equals(statusType) || IN_TRANSIT.equals(statusType))) {
+                    Instant now = Instant.now();
+                    orderRegistrar.orderTest(DSMServer.careEvolveAuth, kit.getHRUID(), kit.getKitLabel(), kit.getExternalOrderNumber(), now);
+                    logger.info("Placed CE order for kit with external order number " + kit.getExternalOrderNumber());
+                }
+                //if delivered notify pepper for received
+                else if (statusType.equals(DELIVERY) && !(DELIVERY.equals(oldType))) {
+                    //                            GBFRequestUtil.updateReceivedDateForKit(kit.getDsmKitRequestId());
+                    logger.info("RECEIVED: " + trackingId);
+                    KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER + SELECT_BY_RETURN_NUMBER, new String[] { RECEIVED, trackingId }, 2);//todo change this to the number of subkits but for now 2 for test boston works
+                    if (kitDDPNotification != null) {
+                        logger.info("Triggering DDP for received kit with external order number: " + kit.getExternalOrderNumber());
+                        EventUtil.triggerDDP(kitDDPNotification);
+
+                    }
+                    else {
+                        logger.error("received kitDDPNotification was null for " + kit.getExternalOrderNumber());
+                    }
+                }
```

Comment:
An `else` here will be helpful, just to log something like "We don't know what to do here".

Reply:
I think the log on the next line explains that. What do you think? Because if we don't know what to do it's because the new status was irrelevant to us and the logs say it

Reply:
Ah, got it!